core/src/main/java/com/linecorp/armeria/client/redirect/RedirectConfigBuilder.java

```diff
@@ -0,0 +1,207 @@
+     * automatic redirection is executed.
+     *
+     * <p>If the {@link WebClient} is created <b>without</b> a base URI, the {@link WebClient} executes
+     * automatic redirection to all domains by default.
+     * If the {@link WebClient} is created <b>with</b> a base URI, automatic redirection is executed to
+     * the domain of the base URI by default.
+     */
+    public RedirectConfigBuilder allowDomains(Iterable<String> domains) {
+        requireNonNull(domains, "domains");
+        if (allowedDomains == null) {
+            allowedDomains = new HashSet<>();
+        }
+        allowedDomains.addAll(ImmutableList.copyOf(domains));
+        return this;
+    }
+
+    /**
+     * Sets the {@link BiPredicate} that returns {@code true} if the {@linkplain URI#getHost() host component}
+     * of a redirection URI is allowed for automatic redirection.
+     *
+     * <p>If the {@link WebClient} is created <b>without</b> a base URI, the {@link WebClient} executes
+     * automatic redirection to all domains by default.
+     * If the {@link WebClient} is created <b>with</b> a base URI, automatic redirection is executed to
+     * the domain of the base URI by default.
+     */
+    public RedirectConfigBuilder allowDomains(
+            BiPredicate<? super ClientRequestContext, ? super String> predicate) {
+        requireNonNull(predicate, "predicate");
+        if (this.predicate == null) {
+            //noinspection unchecked
+            this.predicate = (BiPredicate<ClientRequestContext, String>) predicate;
+        } else {
+            this.predicate = this.predicate.or(predicate);
+        }
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link RedirectConfig} based on the properties set so far.
+     */
+    public RedirectConfig build() {
+        BiPredicate<ClientRequestContext, String> predicate;
+        if (isAllowingAllDomains) {
+            predicate = allowAllDomains;
+        } else {
+            final BiPredicate<ClientRequestContext, String> allowedDomains = allowedDomains();
+            if (allowedDomains != null) {
+                predicate = allowedDomains;
+                if (this.predicate != null) {
+                    predicate = predicate.or(this.predicate);
+                }
+            } else {
+                predicate = this.predicate;
+            }
+        }
+
+        return new RedirectConfig(allowedProtocols, predicate, maxRedirects);
+    }
+
+    @Nullable
+    private BiPredicate<ClientRequestContext, String> allowedDomains() {
+        if (allowedDomains != null) {
+            final Set<String> allowedDomains0 = ImmutableSet.copyOf(allowedDomains);
+            return (ctx, domain) -> {
+                for (String d : allowedDomains0) {
+                    // Use `equals()` to prevent Open Redirects.
+                    // See https://www.acunetix.com/blog/web-security-zone/what-are-open-redirects/
```

Comment:
Looks nice. By the way, don't we need to update the Javadoc?

Reply:
Yeah, let me update it. 😉